javadoc-crawler/build.gradle.kts

```diff
@@ -0,0 +1,34 @@
+plugins {
+  id("otel.java-conventions")
+}
+
+dependencies {
+  implementation("com.fasterxml.jackson.core:jackson-databind")
+  testImplementation("org.assertj:assertj-core:3.27.3")
+}
+
+description = "OpenTelemetry Javadoc Crawler"
+otelJava.moduleName.set("io.opentelemetry.javadocs")
+
+tasks {
+  withType<JavaCompile>().configureEach {
+    sourceCompatibility = "17"
+    targetCompatibility = "17"
```

Comment:
What java 17 features are you using?

Reply:
[text blocks](https://github.com/open-telemetry/opentelemetry-java/pull/7300/files#diff-277d78048c426e4e10e975d5344ec930d3ad6f76498aaaf3279be4c2be616b2dR37-R47) which are 15+

Reply:
this reminded me that we could use a [record](https://github.com/open-telemetry/opentelemetry-java/pull/7300/files#diff-d02bf5e8b9fd231a5da6d6b831a1f165552894ffb8d8de2d88cc3880c18d9b5f) class too

Reply:
Ah, text blocks used in the test. Seems fine, given that 1. this is internal project tooling 2. the project requires builds to use java 17.